tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py

```diff
@@ -189,4 +189,92 @@ def test_call_cudnn_backrward(self):
             self.assertEqual(func.called, self.expect)
 
 
+@testing.parameterize(*testing.product({
+    'c_contiguous': [True, False],
+    'deterministic': [True, False],
+    'nobias': [True, False],
+}))
+class TestConvolution2DFunctionDeterministic(unittest.TestCase):
+    def setUp(self):
+        self.stride = 2
+        self.pad = 1
+        batch_sz = 2
+        in_channels = 64
+        out_channels = 64
+        kh, kw = (3, 3)
+        in_h, in_w = (32, 128)
+        out_h, out_w = (16, 64)
+        # should be same types for cudnn test
+        x_dtype = numpy.float32
+        W_dtype = numpy.float32
+        self.W = numpy.random.normal(
+            0, numpy.sqrt(1. / (kh * kw * in_channels)),
+            (out_channels, in_channels, kh, kw)).astype(W_dtype)
+        self.b = numpy.random.uniform(-1, 1, out_channels).astype(x_dtype)
+        self.x = numpy.random.uniform(
+            -1, 1, (batch_sz, in_channels, in_h, in_w)).astype(x_dtype)
+        self.gy = numpy.random.uniform(
+            -1, 1, (batch_sz, out_channels, out_h, out_w)).astype(x_dtype)
+
+    @attr.gpu
+    @attr.cudnn
+    def test_deterministic(self):
+        """Check deterministic behaviour.
+
+        Result of gradient of W after backward computation might be
+        non-deterministic.
+
+        """
+        x1, W1, b1, y1 = self._run()
+        x2, W2, b2, y2 = self._run()
+
+        cuda.cupy.testing.assert_array_equal(y1.data, y2.data)
+        if self.deterministic:
+            cuda.cupy.testing.assert_array_equal(W1.grad, W2.grad)
+        else:
+            with self.assertRaises(AssertionError):
+                cuda.cupy.testing.assert_array_equal(W1.grad, W2.grad)
```

Comment:
Yes, for `numpy.float32` data types, cudnn's chosen backward filter algorithm is non-deterministic (CUDNN_CONVOLUTION_BWD_FILTER_ALGO_3) and the computed gradients of the weights in the two runs will be slightly different.